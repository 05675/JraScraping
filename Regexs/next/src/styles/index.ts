export { StyledButton } from './button';
export { StyledInput } from './input';
export { StyledLabel } from './label';
export * from './listCommon';
