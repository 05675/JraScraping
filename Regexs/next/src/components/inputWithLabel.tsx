import React from 'react';
import { StyledLabel, StyledErrorText, StyledInput } from '@src/styles';
import { StyledErrorSVG } from '@src/styles/svg';
import { InputWithLabelProps } from '@src/interfaces/lifeInsuranceInputs';

export const InputWithLabel: React.FC<InputWithLabelProps> = props => {
  return (
    <div style={{ marginBottom: '32px' }}>
      <div className='label'>
        <StyledLabel id='titleLabel'>{props.label}</StyledLabel>
      </div>
      <div style={{ position: 'relative' }}>
        <StyledInput id='inputField' {...props} />
        {props.isError && (
          <span className='errorIcon'>
            <StyledErrorSVG id='errorIcon' color='#fa3939' />
          </span>
        )}
      </div>
      {props.isError && (
        <div className='errorText'>
          <StyledErrorText id='errorText'>{props.errorMessage}</StyledErrorText>
        </div>
      )}

      {/* style */}
      <style jsx>{`
        .label {
          height: 18px;
          margin: 0;
        }
        .errorIcon {
          position: absolute;
          width: 24px;
          height: 24px;
          top: 12px;
          right: 0px;
        }
        .errorText {
          width: 100%;
          left: 0px;
          margin-top: 6px;
        }
      `}</style>
    </div>
  );
};
