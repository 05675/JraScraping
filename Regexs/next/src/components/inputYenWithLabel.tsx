import React from 'react';
import { StyledLabel, StyledErrorText, StyledInput } from '@src/styles';
import { StyledErrorSVG } from '@src/styles/svg';
import { InputWithLabelProps } from '@src/interfaces/lifeInsuranceInputs';

export const InputYenWithLabel: React.FC<InputWithLabelProps> = props => {
  return (
    <div style={{ marginBottom: '32px' }}>
      <div className='label'>
        <StyledLabel id='titleLabel'>{props.label}</StyledLabel>
      </div>
      <div style={{ position: 'relative' }}>
        <StyledInput id='inputField' {...props} isPaddingRight />
        <span className='yenChar'>
          <StyledLabel id='yenLabel'>å††</StyledLabel>
        </span>
        {props.isError && (
          <span className='errorIcon'>
            <StyledErrorSVG id='errorIcon' color='#fa3939' />
          </span>
        )}
      </div>
      {props.isError && (
        <div className='errorText'>
          <StyledErrorText id='errorText'>{props.errorMessage}</StyledErrorText>
        </div>
      )}

      {/* style */}
      <style jsx>{`
        .label {
          height: 18px;
          margin: 0;
        }
        .yenChar {
          position: absolute;
          width: 12px;
          height: 18px;
          bottom: 18px;
          right: ${props.isError ? '40px' : '0px'};
        }
        .errorIcon {
          position: absolute;
          width: 24px;
          height: 24px;
          top: 12px;
          right: 0px;
        }
        .errorText {
          width: 100%;
          left: 0px;
          margin-top: 6px;
        }
      `}</style>
    </div>
  );
};
