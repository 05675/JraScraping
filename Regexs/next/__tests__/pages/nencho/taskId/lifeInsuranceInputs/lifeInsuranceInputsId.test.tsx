/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { InputWithLabel } from '@src/components/inputWithLabel';
import { SelectBoxWithLabel } from '@src/components/selectBoxWithLabel';
import {
  LifeInsuranceInputsProps,
  NenchoInsuranceLifeInput2020,
} from '@src/interfaces/lifeInsuranceInputs';
import {
  LifeInsuranceInputs,
  categoryOptions,
} from '@src/pages/nencho/[taskId]/lifeInsuranceInputs/[lifeInsuranceInputsId]';

import axios from 'axios';

import Router from 'next/router';
import { StyledButton } from '@src/styles';
import { errorMsg } from '@src/util/constantMsg';
import { InputYenWithLabel } from '@src/components/inputYenWithLabel';

jest.mock('axios');
(axios.put as any).mockImplementation(() => Promise.resolve({ status: 200, data: {} }));
(axios.post as any).mockImplementation(() => Promise.resolve({ status: 200, data: {} }));
(axios.get as any).mockImplementation(() => Promise.resolve({ status: 200, data: {} }));
(axios.delete as any).mockImplementation(() => Promise.resolve({ status: 200, data: {} }));

Router.push = jest.fn();

const lifeInsuranceDataDefault: NenchoInsuranceLifeInput2020 = {
  nenchoId: 'nenchoIdData',
  category: { id: 'insurance_0001', name: 'category' },
  firmName: 'firmNameData',
  categoryDetail: 'categoryDetailData',
  period: 'periodData',
  contractorName: 'contractorNameData',
  receiverName: 'receiverNameData',
  relation: 'relationData',
  payment: 'paymentData',
};

const propsDefault: LifeInsuranceInputsProps = {
  isUpdatePage: true,
  nenchoId: 'nenchoIdDummy',
  lifeInsuranceData: lifeInsuranceDataDefault,
};

const propsBlank: LifeInsuranceInputsProps = {
  isUpdatePage: false,
  nenchoId: 'nenchoIdDummy',
};

describe('snapshot', () => {
  test('default', () => {
    const snapshot = renderer.create(<LifeInsuranceInputs {...propsDefault} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});

describe('button 保存', () => {
  test('初期表示', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const target = nextPage.find(StyledButton).find('[name="saveLifeInsurance"]');
    expect(target.length).toBe(1);

    expect(target.text()).toMatch('保存');
    expect(target.prop('important')).toBeTruthy();
    expect(target.prop('smallSize')).toBeTruthy();
    expect(target.prop('onClick')?.name).toMatch('handleClickForUpsert');
  });
});

describe('SelectBoxWithLabel 加入している生命保険', () => {
  test('初期表示：初期値なし', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const target = nextPage.find(SelectBoxWithLabel).find('[name="category"]');
    expect(target.length).toBe(1);

    expect(target.prop('label')).toMatch('加入している生命保険');
    expect(target.prop('value')).toMatch('');
    expect(target.prop('options')).toEqual(categoryOptions);
    expect(target.prop('onChange')?.name).toMatch('handleChangeForSelectInsuranceCategory');
    expect(target.prop('isError')).toBeFalsy();
    expect(target.prop('errorMessage')).toMatch('');
  });
  test('初期表示：初期値あり', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsDefault} />);
    const target = nextPage.find(SelectBoxWithLabel).find('[name="category"]');
    expect(target.length).toBe(1);

    expect(target.prop('value')).toMatch('insurance_0001');
  });
  test('バリデーションエラー：未入力', () => {
    const nextPagePre = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const saveInputButton = nextPagePre.find(StyledButton).find('[name="saveLifeInsurance"]');
    expect(saveInputButton.length).toBe(1);
    saveInputButton.simulate('click');

    // Stateが更新されたのでupdateが必要
    const nextPage = nextPagePre.update();
    const target = nextPage.find(SelectBoxWithLabel).find('[name="category"]');
    expect(target.length).toBe(1);

    expect(target.prop('isError')).toBeTruthy();
    expect(target.prop('errorMessage')).toMatch(errorMsg.VALIDATE_ERR_EMPTY_CATEGORY);
  });
});

describe('InputWithLabel 保険会社等の名称', () => {
  test('初期表示：初期値なし', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const target = nextPage.find(InputWithLabel).find('[name="firmName"]');
    expect(target.length).toBe(1);

    expect(target.prop('label')).toMatch('保険会社等の名称');
    expect(target.prop('value')).toMatch('');
    expect(target.prop('onChange')?.name).toMatch('handleChange');
    expect(target.prop('isError')).toBeFalsy();
    expect(target.prop('errorMessage')).toMatch('');
    expect(target.prop('maxLength')).toBe(36);
  });
  test('初期表示：初期値あり', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsDefault} />);
    const target = nextPage.find(InputWithLabel).find('[name="firmName"]');
    expect(target.length).toBe(1);

    expect(target.prop('value')).toMatch('firmNameData');
  });
  test('バリデーションエラー：未入力', () => {
    const nextPagePre = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const saveInputButton = nextPagePre.find(StyledButton).find('[name="saveLifeInsurance"]');
    expect(saveInputButton.length).toBe(1);
    saveInputButton.simulate('click');

    // Stateが更新されたのでupdateが必要
    const nextPage = nextPagePre.update();
    const target = nextPage.find(InputWithLabel).find('[name="firmName"]');
    expect(target.length).toBe(1);

    expect(target.prop('isError')).toBeTruthy();
    expect(target.prop('errorMessage')).toMatch(errorMsg.VALIDATE_ERR_EMPTY_FIRM_NAME);
  });
});

describe('InputWithLabel 保険等の種類', () => {
  test('初期表示：初期値なし', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const target = nextPage.find(InputWithLabel).find('[name="categoryDetail"]');
    expect(target.length).toBe(1);

    expect(target.prop('label')).toMatch('保険等の種類');
    expect(target.prop('value')).toMatch('');
    expect(target.prop('onChange')?.name).toMatch('handleChange');
    expect(target.prop('isError')).toBeFalsy();
    expect(target.prop('errorMessage')).toMatch('');
    expect(target.prop('maxLength')).toBe(36);
  });
  test('初期表示：初期値あり', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsDefault} />);
    const target = nextPage.find(InputWithLabel).find('[name="categoryDetail"]');
    expect(target.length).toBe(1);

    expect(target.prop('value')).toMatch('categoryDetailData');
  });
  test('バリデーションエラー：未入力', () => {
    const nextPagePre = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const saveInputButton = nextPagePre.find(StyledButton).find('[name="saveLifeInsurance"]');
    expect(saveInputButton.length).toBe(1);
    saveInputButton.simulate('click');

    // Stateが更新されたのでupdateが必要
    const nextPage = nextPagePre.update();
    const target = nextPage.find(InputWithLabel).find('[name="categoryDetail"]');
    expect(target.length).toBe(1);

    expect(target.prop('isError')).toBeTruthy();
    expect(target.prop('errorMessage')).toMatch(errorMsg.VALIDATE_ERR_EMPTY_CATEGORY_DETAIL);
  });
});

describe('InputWithLabel 保険期間', () => {
  test('初期表示：初期値なし', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const target = nextPage.find(InputWithLabel).find('[name="period"]');
    expect(target.length).toBe(1);

    expect(target.prop('label')).toMatch('保険期間');
    expect(target.prop('value')).toMatch('');
    expect(target.prop('onChange')?.name).toMatch('handleChange');
    expect(target.prop('isError')).toBeFalsy();
    expect(target.prop('errorMessage')).toMatch('');
    expect(target.prop('maxLength')).toBe(36);
  });
  test('初期表示：初期値あり', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsDefault} />);
    const target = nextPage.find(InputWithLabel).find('[name="period"]');
    expect(target.length).toBe(1);

    expect(target.prop('value')).toMatch('periodData');
  });
  test('バリデーションエラー：未入力', () => {
    const nextPagePre = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const saveInputButton = nextPagePre.find(StyledButton).find('[name="saveLifeInsurance"]');
    expect(saveInputButton.length).toBe(1);
    saveInputButton.simulate('click');

    // Stateが更新されたのでupdateが必要
    const nextPage = nextPagePre.update();
    const target = nextPage.find(InputWithLabel).find('[name="period"]');
    expect(target.length).toBe(1);

    expect(target.prop('isError')).toBeTruthy();
    expect(target.prop('errorMessage')).toMatch(errorMsg.VALIDATE_ERR_EMPTY_PERIOD);
  });
});

describe('InputWithLabel 保険等の契約者の氏名', () => {
  test('初期表示：初期値なし', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const target = nextPage.find(InputWithLabel).find('[name="contractorName"]');
    expect(target.length).toBe(1);

    expect(target.prop('label')).toMatch('保険等の契約者の氏名');
    expect(target.prop('value')).toMatch('');
    expect(target.prop('onChange')?.name).toMatch('handleChange');
    expect(target.prop('isError')).toBeFalsy();
    expect(target.prop('errorMessage')).toMatch('');
    expect(target.prop('maxLength')).toBe(36);
  });
  test('初期表示：初期値あり', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsDefault} />);
    const target = nextPage.find(InputWithLabel).find('[name="contractorName"]');
    expect(target.length).toBe(1);

    expect(target.prop('value')).toMatch('contractorNameData');
  });
  test('バリデーションエラー：未入力', () => {
    const nextPagePre = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const saveInputButton = nextPagePre.find(StyledButton).find('[name="saveLifeInsurance"]');
    expect(saveInputButton.length).toBe(1);
    saveInputButton.simulate('click');

    // Stateが更新されたのでupdateが必要
    const nextPage = nextPagePre.update();
    const target = nextPage.find(InputWithLabel).find('[name="contractorName"]');
    expect(target.length).toBe(1);

    expect(target.prop('isError')).toBeTruthy();
    expect(target.prop('errorMessage')).toMatch(errorMsg.VALIDATE_ERR_EMPTY_CONTRACTOR_NAME);
  });
});

describe('InputWithLabel 保険金等の受取人 氏名', () => {
  test('初期表示：初期値なし', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const target = nextPage.find(InputWithLabel).find('[name="receiverName"]');
    expect(target.length).toBe(1);

    expect(target.prop('label')).toMatch('氏名');
    expect(target.prop('value')).toMatch('');
    expect(target.prop('onChange')?.name).toMatch('handleChange');
    expect(target.prop('isError')).toBeFalsy();
    expect(target.prop('errorMessage')).toMatch('');
    expect(target.prop('maxLength')).toBe(36);
  });
  test('初期表示：初期値あり', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsDefault} />);
    const target = nextPage.find(InputWithLabel).find('[name="receiverName"]');
    expect(target.length).toBe(1);

    expect(target.prop('value')).toMatch('receiverNameData');
  });
  test('バリデーションエラー：未入力', () => {
    const nextPagePre = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const saveInputButton = nextPagePre.find(StyledButton).find('[name="saveLifeInsurance"]');
    expect(saveInputButton.length).toBe(1);
    saveInputButton.simulate('click');

    // Stateが更新されたのでupdateが必要
    const nextPage = nextPagePre.update();
    const target = nextPage.find(InputWithLabel).find('[name="receiverName"]');
    expect(target.length).toBe(1);

    expect(target.prop('isError')).toBeTruthy();
    expect(target.prop('errorMessage')).toMatch(errorMsg.VALIDATE_ERR_EMPTY_RECEIVER_NAME);
  });
});

describe('InputWithLabel 保険金等の受取人 あなたとの続柄', () => {
  test('初期表示：初期値なし', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const target = nextPage.find(InputWithLabel).find('[name="relation"]');
    expect(target.length).toBe(1);

    expect(target.prop('label')).toMatch('あなたとの続柄');
    expect(target.prop('value')).toMatch('');
    expect(target.prop('onChange')?.name).toMatch('handleChange');
    expect(target.prop('isError')).toBeFalsy();
    expect(target.prop('errorMessage')).toMatch('');
    expect(target.prop('maxLength')).toBe(6);
  });
  test('初期表示：初期値あり', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsDefault} />);
    const target = nextPage.find(InputWithLabel).find('[name="relation"]');
    expect(target.length).toBe(1);

    expect(target.prop('value')).toMatch('relationData');
  });
  test('バリデーションエラー：未入力', () => {
    const nextPagePre = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const saveInputButton = nextPagePre.find(StyledButton).find('[name="saveLifeInsurance"]');
    expect(saveInputButton.length).toBe(1);
    saveInputButton.simulate('click');

    // Stateが更新されたのでupdateが必要
    const nextPage = nextPagePre.update();
    const target = nextPage.find(InputWithLabel).find('[name="relation"]');
    expect(target.length).toBe(1);

    expect(target.prop('isError')).toBeTruthy();
    expect(target.prop('errorMessage')).toMatch(errorMsg.VALIDATE_ERR_EMPTY_RELATION);
  });
});

describe('InputWithLabel 申請額', () => {
  test('初期表示：初期値なし', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const target = nextPage.find(InputYenWithLabel).find('[name="payment"]');
    expect(target.length).toBe(1);

    expect(target.prop('label')).toMatch('申請額');
    expect(target.prop('value')).toMatch('');
    expect(target.prop('onChange')?.name).toMatch('handleChange');
    expect(target.prop('isError')).toBeFalsy();
    expect(target.prop('errorMessage')).toMatch('');
    expect(target.prop('maxLength')).toBe(8);
    expect(target.prop('type')).toMatch('tel');
  });
  test('初期表示：初期値あり', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsDefault} />);
    const target = nextPage.find(InputYenWithLabel).find('[name="payment"]');
    expect(target.length).toBe(1);

    expect(target.prop('value')).toMatch('paymentData');
  });
  test('バリデーションエラー：未入力', () => {
    const nextPagePre = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const saveInputButton = nextPagePre.find(StyledButton).find('[name="saveLifeInsurance"]');
    expect(saveInputButton.length).toBe(1);
    saveInputButton.simulate('click');

    // Stateが更新されたのでupdateが必要
    const nextPage = nextPagePre.update();
    const target = nextPage.find(InputYenWithLabel).find('[name="payment"]');
    expect(target.length).toBe(1);

    expect(target.prop('isError')).toBeTruthy();
    expect(target.prop('errorMessage')).toMatch(errorMsg.VALIDATE_ERR_EMPTY_PAYMENT);
  });
  test('入力制限：数字10桁', () => {
    let nextPage = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const saveInputButton = nextPage.find(StyledButton).find('[name="saveLifeInsurance"]');
    expect(saveInputButton.length).toBe(1);
    let target = nextPage.find(InputYenWithLabel).find('[name="payment"]');
    expect(target.length).toBe(1);

    expect(target.prop('value')).toMatch('');

    target.simulate('change', {
      target: { name: 'payment', value: '1234567890' },
    });
    nextPage = nextPage.update();
    target = nextPage.find(InputYenWithLabel).find('[name="payment"]');
    expect(target.length).toBe(1);
    expect(target.prop('value')).toMatch('1234567890');
  });
  test('入力制限：数字11桁', () => {
    let nextPage = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const saveInputButton = nextPage.find(StyledButton).find('[name="saveLifeInsurance"]');
    expect(saveInputButton.length).toBe(1);
    let target = nextPage.find(InputYenWithLabel).find('[name="payment"]');
    expect(target.length).toBe(1);

    expect(target.prop('value')).toMatch('');

    target.simulate('change', {
      target: { name: 'payment', value: '12345678901' },
    });
    nextPage = nextPage.update();
    target = nextPage.find(InputYenWithLabel).find('[name="payment"]');
    expect(target.length).toBe(1);
    expect(target.prop('value')).toMatch('1234567890');
  });
  test('入力制限：数字以外', () => {
    let nextPage = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const saveInputButton = nextPage.find(StyledButton).find('[name="saveLifeInsurance"]');
    expect(saveInputButton.length).toBe(1);
    let target = nextPage.find(InputYenWithLabel).find('[name="payment"]');
    expect(target.length).toBe(1);

    expect(target.prop('value')).toMatch('');

    target.simulate('change', {
      target: { name: 'payment', value: '001*2あ3ア4ｱ5a6A7' },
    });
    nextPage = nextPage.update();
    target = nextPage.find(InputYenWithLabel).find('[name="payment"]');
    expect(target.length).toBe(1);
    expect(target.prop('value')).toMatch('1234567');
  });
});

describe('button この保険を削除する', () => {
  test('初期表示：isUpdatePage=false', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsBlank} />);
    const target = nextPage.find(StyledButton).find('[name="deleteLifeInsurance"]');
    expect(target.length).toBe(0);
  });
  test('初期表示：isUpdatePage=true', () => {
    const nextPage = shallow(<LifeInsuranceInputs {...propsDefault} />);
    const target = nextPage.find(StyledButton).find('[name="deleteLifeInsurance"]');
    expect(target.length).toBe(1);

    expect(target.text()).toMatch('この保険を削除する');
    expect(target.prop('warning')).toBeTruthy();
    expect(target.prop('onClick')?.name).toMatch('confirm');
  });
});
