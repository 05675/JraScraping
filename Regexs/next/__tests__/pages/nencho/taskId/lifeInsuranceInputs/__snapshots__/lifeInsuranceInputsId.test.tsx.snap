// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot default 1`] = `
<div
  style={
    Object {
      "background": "#FFFFFF",
    }
  }
>
  <div
    className="page-header page-header-life-insurance headroom-wrapper"
    style={
      Object {
        "height": null,
      }
    }
  >
    <div
      className="headroom headroom--unfixed"
    >
      <div
        style={
          Object {
            "background": "#DEEEFC",
            "position": "fixed",
            "width": "100%",
            "zIndex": 1,
          }
        }
      >
        <div
          style={
            Object {
              "background": "#FFFFFF",
              "borderRadius": "8px 8px 0px 0px",
              "marginTop": "16px",
            }
          }
        >
          <button
            className="sc-AxjAm GLyvs"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#ffffff",
                "color": "#525252",
                "height": "12px",
                "left": "16px",
                "padding": "0",
                "position": "absolute",
                "top": "38px",
              }
            }
            width="65px"
          >
            キャンセル
          </button>
          <title
            className="sc-AxhUy dpKhk"
          >
            生命保険
          </title>
          <button
            className="sc-AxjAm dxkTwa"
            disabled={true}
            name="saveLifeInsurance"
            onClick={[Function]}
            style={
              Object {
                "position": "absolute",
                "right": "16px",
                "top": "32px",
              }
            }
            width="62px"
          >
            保存
          </button>
        </div>
      </div>
      <div
        style={
          Object {
            "background": "#FFFFFF",
            "height": "80px",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div
    style={
      Object {
        "height": "128px",
      }
    }
  />
  <div
    style={
      Object {
        "margin": "0px 16px",
      }
    }
  >
    <div
      style={
        Object {
          "marginBottom": "32px",
        }
      }
    >
      <div
        className="label"
      >
        <label
          className="sc-AxhCb fZxPgc"
          id="titleLabel"
        >
          加入している生命保険
        </label>
      </div>
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <div
          className=" css-2b097c-container"
          id="selectBoxSingle"
          onKeyDown={[Function]}
        >
          <div
            className=" css-88o9p8-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-86rg1c-ValueContainer"
            >
              <div
                className=" css-1neebod-singleValue"
              >
                一般の生命保険：新制度
              </div>
              <input
                aria-autocomplete="list"
                className="css-62g3xt-dummyInput"
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                readOnly={true}
                tabIndex="0"
                value=""
              />
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-62wm9a-IndicatorSeparator"
              />
              <div
                style={
                  Object {
                    "marginRight": "0",
                  }
                }
              >
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg>

                    className="sc-fzokOt enYZSd"
                    color="#949494"
                    height="12px"
                    width="12px"
                  />
                </div>
              </div>
            </div>
          </div>
          <input
            name="category"
            type="hidden"
            value="insurance_0001"
          />
        </div>
      </div>
      <style
        jsx={true}
      >
        
        .label {
          height: 18px;
          margin: 0;
        }
        .errorIcon {
          position: absolute;
          width: 24px;
          height: 24px;
          top: 12px;
          right: 0px;
          background: white;
        }
        .errorText {
          width: 100%;
          left: 0px;
          margin-top: 6px;
        }
      
      </style>
    </div>
    <div
      style={
        Object {
          "marginBottom": "32px",
        }
      }
    >
      <div
        className="label"
      >
        <label
          className="sc-AxhCb fZxPgc"
          id="titleLabel"
        >
          保険会社等の名称
        </label>
      </div>
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <input
          className="sc-AxiKw cyBugS"
          id="inputField"
          label="保険会社等の名称"
          maxLength={36}
          name="firmName"
          onChange={[Function]}
          value="firmNameData"
        />
      </div>
      <style
        jsx={true}
      >
        
        .label {
          height: 18px;
          margin: 0;
        }
        .errorIcon {
          position: absolute;
          width: 24px;
          height: 24px;
          top: 12px;
          right: 0px;
        }
        .errorText {
          width: 100%;
          left: 0px;
          margin-top: 6px;
        }
      
      </style>
    </div>
    <div
      style={
        Object {
          "marginBottom": "32px",
        }
      }
    >
      <div
        className="label"
      >
        <label
          className="sc-AxhCb fZxPgc"
          id="titleLabel"
        >
          保険等の種類
        </label>
      </div>
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <input
          className="sc-AxiKw cyBugS"
          id="inputField"
          label="保険等の種類"
          maxLength={36}
          name="categoryDetail"
          onChange={[Function]}
          value="categoryDetailData"
        />
      </div>
      <style
        jsx={true}
      >
        
        .label {
          height: 18px;
          margin: 0;
        }
        .errorIcon {
          position: absolute;
          width: 24px;
          height: 24px;
          top: 12px;
          right: 0px;
        }
        .errorText {
          width: 100%;
          left: 0px;
          margin-top: 6px;
        }
      
      </style>
    </div>
    <div
      style={
        Object {
          "marginBottom": "32px",
        }
      }
    >
      <div
        className="label"
      >
        <label
          className="sc-AxhCb fZxPgc"
          id="titleLabel"
        >
          保険期間
        </label>
      </div>
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <input
          className="sc-AxiKw cyBugS"
          id="inputField"
          label="保険期間"
          maxLength={36}
          name="period"
          onChange={[Function]}
          value="periodData"
        />
      </div>
      <style
        jsx={true}
      >
        
        .label {
          height: 18px;
          margin: 0;
        }
        .errorIcon {
          position: absolute;
          width: 24px;
          height: 24px;
          top: 12px;
          right: 0px;
        }
        .errorText {
          width: 100%;
          left: 0px;
          margin-top: 6px;
        }
      
      </style>
    </div>
    <div
      style={
        Object {
          "marginBottom": "32px",
        }
      }
    >
      <div
        className="label"
      >
        <label
          className="sc-AxhCb fZxPgc"
          id="titleLabel"
        >
          保険等の契約者の氏名
        </label>
      </div>
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <input
          className="sc-AxiKw cyBugS"
          id="inputField"
          label="保険等の契約者の氏名"
          maxLength={36}
          name="contractorName"
          onChange={[Function]}
          value="contractorNameData"
        />
      </div>
      <style
        jsx={true}
      >
        
        .label {
          height: 18px;
          margin: 0;
        }
        .errorIcon {
          position: absolute;
          width: 24px;
          height: 24px;
          top: 12px;
          right: 0px;
        }
        .errorText {
          width: 100%;
          left: 0px;
          margin-top: 6px;
        }
      
      </style>
    </div>
    <label
      className="sc-AxhCb fZxPgc"
    >
      保険金等の受取人
    </label>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <div
          style={
            Object {
              "marginBottom": "32px",
            }
          }
        >
          <div
            className="label"
          >
            <label
              className="sc-AxhCb fZxPgc"
              id="titleLabel"
            >
              氏名
            </label>
          </div>
          <div
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <input
              className="sc-AxiKw cyBugS"
              id="inputField"
              label="氏名"
              maxLength={36}
              name="receiverName"
              onChange={[Function]}
              value="receiverNameData"
            />
          </div>
          <style
            jsx={true}
          >
            
        .label {
          height: 18px;
          margin: 0;
        }
        .errorIcon {
          position: absolute;
          width: 24px;
          height: 24px;
          top: 12px;
          right: 0px;
        }
        .errorText {
          width: 100%;
          left: 0px;
          margin-top: 6px;
        }
      
          </style>
        </div>
      </div>
      <div
        style={
          Object {
            "marginLeft": "16px",
            "width": "50%",
          }
        }
      >
        <div
          style={
            Object {
              "marginBottom": "32px",
            }
          }
        >
          <div
            className="label"
          >
            <label
              className="sc-AxhCb fZxPgc"
              id="titleLabel"
            >
              あなたとの続柄
            </label>
          </div>
          <div
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <input
              className="sc-AxiKw cyBugS"
              id="inputField"
              label="あなたとの続柄"
              maxLength={6}
              name="relation"
              onChange={[Function]}
              value="relationData"
            />
          </div>
          <style
            jsx={true}
          >
            
        .label {
          height: 18px;
          margin: 0;
        }
        .errorIcon {
          position: absolute;
          width: 24px;
          height: 24px;
          top: 12px;
          right: 0px;
        }
        .errorText {
          width: 100%;
          left: 0px;
          margin-top: 6px;
        }
      
          </style>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "marginBottom": "32px",
        }
      }
    >
      <div
        className="label"
      >
        <label
          className="sc-AxhCb fZxPgc"
          id="titleLabel"
        >
          申請額
        </label>
      </div>
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <input
          className="sc-AxiKw gHDYXa"
          id="inputField"
          label="申請額"
          maxLength={10}
          name="payment"
          onChange={[Function]}
          type="tel"
          value="paymentData"
        />
        <span
          className="yenChar"
        >
          <label
            className="sc-AxhCb fZxPgc"
            id="yenLabel"
          >
            円
          </label>
        </span>
      </div>
      <style
        jsx={true}
      >
        
        .label {
          height: 18px;
          margin: 0;
        }
        .yenChar {
          position: absolute;
          width: 12px;
          height: 18px;
          bottom: 18px;
          right: 0px;
        }
        .errorIcon {
          position: absolute;
          width: 24px;
          height: 24px;
          top: 12px;
          right: 0px;
        }
        .errorText {
          width: 100%;
          left: 0px;
          margin-top: 6px;
        }
      
      </style>
    </div>
    <div
      className="sc-AxgMl hnSpCS"
      style={
        Object {
          "paddingBottom": "32px",
        }
      }
    >
      あなたが本年中に支払った保険料等の金額（分配を受けた剰余金等の控除後の金額）
    </div>
    <button
      className="sc-AxjAm ebJHzb"
      name="deleteLifeInsurance"
      onClick={[Function]}
      style={
        Object {
          "marginBottom": "32px",
          "width": "100%",
        }
      }
    >
      この保険を削除する
    </button>
  </div>
  <div
    className="notif-container"
  >
    <p>
      
    </p>
    <span
      className="close-button"
      onClick={[Function]}
      role="button"
    >
      <svg>

        className="sc-fzoLag gAJUJy"
        color="#ffffff"
        height="14"
        width="14"
      />
    </span>
    <style
      jsx={true}
    >
      
            .notif-container {
              position: fixed;
              width: 100%;
              height: 48px;
              bottom: 0;
              display: flex;
              box-shadow: 0px -4px 8px rgba(0, 0, 0, 0.2);
              border-radius: 8px 8px 0px 0px;
              align-items: center;
              background: red;
              text-align-center;
              transform: translateY(100%);
              opacity: 0;
              pointer-events: none;
              transition: transform 0.2s ease-in-out, opacity 0.23s ease-in-out;
            }
            .notif-container p {
              font-style: normal;
              font-weight: normal;
              font-size: 12px;
              line-height: 100%;
              letter-spacing: 0.06em;
              color: #ffffff;
              padding-left:16px;
              width: 100%;
            }
            .close-button {
              position: relative;
              padding:0;
              margin-right:13px;
            }
          
    </style>
  </div>
</div>
`;
